README.md

```diff
@@ -205,6 +206,69 @@ iex> {:ok, s3} = MLLP.Sender.start_link("localhost", 8154, tls: [verify: :verify
 iex> MLLP.Sender.send_hl7_and_receive_ack(s3, HL7.Examples.wikipedia_sample_hl7() |> HL7.Message.new())
 ```
 
+## Using Client Certificates
+MLLP listener can force client to provide a valid certificate before establishing a successful connections. Follow steps below to use client certs with listener
+
+### Start MLLP listener with :verify_peer option
+```
+iex> MLLP.Receiver.start(port: 8154, dispatcher: MLLP.EchoDispatcher, transport_opts: %{tls: [cacertfile: "tls/root-ca/ca_certificate.pem", verify: :verify_peer, certfile: "tls/server/server_certificate.pem", keyfile: "tls/server/private_key.pem"]})
+```
+
+### Start MLLP Sender with client cert
+```
+iex> {:ok, s3} = MLLP.Sender.start_link("localhost", 8154, tls: [verify: :verify_peer, cacertfile: "tls/root-ca/ca_certificate.pem", certfile: "tls/client/client_certificate.pem", keyfile: "tls/client/private_key.pem"])
+```
+
+### Send a message
+```
+iex> MLLP.Sender.send_hl7_and_receive_ack(s3, HL7.Examples.wikipedia_sample_hl7() |> HL7.Message.new())
+```
+
+## Using Client Restrictions
+MLLP listener supports two options to restrict incoming client connections to make sure it accepts only trusted clients.
+
+1) IP/DNS restriction - In this mode, we can restrict incoming connections using Client IP/DNS.
+2) Client Cert Check - If `verify: :verify_peer` option is enabled on listener, it will enforce client to send a valid client cert and will only allow connection if it trusts the certificate issued to.
```

Comment:
```suggestion
2) Client Cert Check - If `verify: :verify_peer` option is enabled on the listener, it will enforce the client to send a valid client cert and will only allow the connection if certificate returned from the client is valid and trusted.
```